### PR TITLE
Aniket/dbschemas

### PR DESCRIPTION
**Pull request by AniketLokare**

### CODE REVIEW

File: DBSchema/users_address.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  `USER_ID` int DEFAULT NULL,
+  `Address_Type` enum('Primary','Office') DEFAULT NULL,
+  `Building_Name` varchar(100) DEFAULT NULL,
+  `Building_No` int DEFAULT NULL,
+  `Street` varchar(100) DEFAULT NULL,
+  `City` varchar(50) DEFAULT NULL,
+  `State` varchar(50) NOT NULL,
+  `Pincode` int NOT NULL,
+  `Country` varchar(50) NOT NULL,
+  `Hobbies` set('Reading','Traveling','Cooking','Hiking','Music','Sports') DEFAULT NULL,
```
Comment (ravs788) [inline DBSchema/users_address.sql:36]: Is this  a fixed set of values for now? Will we be treating this as a dropdown in the future?

File: DBSchema/users_address.sql [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  `Address_ID` int NOT NULL AUTO_INCREMENT,
+  `USER_ID` int DEFAULT NULL,
+  `Address_Type` enum('Primary','Office') DEFAULT NULL,
+  `Building_Name` varchar(100) DEFAULT NULL,
+  `Building_No` int DEFAULT NULL,
+  `Street` varchar(100) DEFAULT NULL,
+  `City` varchar(50) DEFAULT NULL,
+  `State` varchar(50) NOT NULL,
+  `Pincode` int NOT NULL,
+  `Country` varchar(50) NOT NULL,
```
Comment (ravs788) [inline DBSchema/users_address.sql:35]: Can we make this as a dropdown in the UI referring to some external API for getting the list? Will we need to make any changes here for the same?

**Review by ravs788 — CHANGES_REQUESTED** [review]
Provided comments appropriately

File: DBSchema/users_address.sql (hunk 1 above)
Comment (AniketLokare) [reply DBSchema/users_address.sql:36]: Yes, this is the fixed set of values. We can add more values in db schemas if we want to showcase this in dropdown

File: DBSchema/users_address.sql (hunk 2 above)
Comment (AniketLokare) [reply DBSchema/users_address.sql:35]: We can manipulate it in the code to store particular format in db

File: DBSchema/users_address.sql (hunk 1 above)
Comment (ravs788) [reply DBSchema/users_address.sql:36]: I think we should keep it a free-form text, we should code to auto-suggest based on values in DB. Maybe you can sync on this with Alind once. Also, I think he meant this field to a be a multi-select, meaning we should be able to have more than 1 value. Do we need to make any db change for that?

Comment (AniketLokare) [reply DBSchema/users_address.sql:36]: I already had discussion with Alind regarding the same. Below are some challenges to store hobbies with free-form text.

1) redundancy of data 
Ex. If 2 users enters singing 2 times, will you take it as 2 hobbies? and then again you maintain mapping of users and hobbies, that will be tedious task.

2) We can store comma separated values as a string for user but in that case all coding tasks like manipulation of the strings and storing it into db will be our headache.

3) We can introduce joins in our table but it also a complex operations for only hobbies thing

4) (Our solution) After All above discussion, we have decided to use set for simple storing.

Comment (ravs788) [reply DBSchema/users_address.sql:36]: Ok then we should be good for now. We will look into this later.

**Review by ravs788 — APPROVED** [review]

**Review by MadhupAtri — APPROVED** [review]
Looks good as of now

**Pull request merged**